staging/src/k8s.io/kubectl/pkg/cmd/exec/exec.go

```diff
@@ -326,10 +326,11 @@ func (p *ExecOptions) Run() error {
 	containerName := p.ContainerName
 	if len(containerName) == 0 {
 		if len(pod.Spec.Containers) > 1 {
-			fmt.Fprintf(p.ErrOut, "Defaulting container name to %s.\n", pod.Spec.Containers[0].Name)
 			if p.EnableSuggestedCmdUsage {
-				fmt.Fprintf(p.ErrOut, "Use '%s describe pod/%s -n %s' to see all of the containers in this pod.\n", p.ParentCommandName, pod.Name, p.Namespace)
+				containerNames := polymorphichelpers.GetContainerNames(pod.Spec.Containers)
```

Comment:
There are other containers may be init-containers and ephemeral-containers could be excuted.

Reply:
yes, I ignored the init-containers and ephemeral-containers. I will add a new function to list both of them, and usually most of pods do not exist  init-containers and ephemeral-containers. so in this function, if both of them do not exist,  their lists print will not be shown.